# Data cleaning and preprocessing

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

STATE = 42

In [2]:
hurricanes = pd.read_csv("data/hurricane_data.csv") # renamed kaggle csv from storms to hurricane_data

## Check df info

In [3]:
hurricanes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22705 entries, 0 to 22704
Data columns (total 13 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   name                          22705 non-null  object 
 1   year                          22705 non-null  int64  
 2   month                         22705 non-null  int64  
 3   day                           22705 non-null  int64  
 4   hour                          22705 non-null  int64  
 5   lat                           22705 non-null  float64
 6   long                          22705 non-null  float64
 7   status                        22705 non-null  object 
 8   category                      5162 non-null   float64
 9   wind                          22705 non-null  int64  
 10  pressure                      22705 non-null  int64  
 11  tropicalstorm_force_diameter  22705 non-null  int64  
 12  hurricane_force_diameter      22705 non-null  int64  
dtypes

We can see that the only column that contains missing values is `category`, which makes sense because the storms in the dataset that are not hurricanes would not have a category.
<br>
<br>
To cross reference, rows that contain the 'hurricane' status will be checked if they contain any missing values from the category columns.

In [4]:
# Extract rows that have a status of "hurricane"
status_hurricane = hurricanes[hurricanes['status'] == 'hurricane'].copy()
print("Difference between all rows of status \"hurricane\" and rows that contain missing values: ", len(status_hurricane) - len(status_hurricane.dropna()))

Difference between all rows of status "hurricane" and rows that contain missing values:  0


We've now accounted for all of the missing values that exist and will be moving forward with the 5162 rows that have the "hurricane" status.

## Check for duplicate values

In [5]:
status_hurricane.duplicated()

70       False
71       False
72       False
73       False
74       False
         ...  
22674    False
22675    False
22676    False
22677    False
22678    False
Length: 5162, dtype: bool

We can see that there are no duplicate values after running the `duplicated` function.

## Create binary Non/Major target variable

In [6]:
status_hurricane['hurricane_class'] = 0
status_hurricane.loc[status_hurricane['category'] > 2.0, 'hurricane_class'] = 1

status_hurricane.info()

<class 'pandas.core.frame.DataFrame'>
Index: 5162 entries, 70 to 22678
Data columns (total 14 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   name                          5162 non-null   object 
 1   year                          5162 non-null   int64  
 2   month                         5162 non-null   int64  
 3   day                           5162 non-null   int64  
 4   hour                          5162 non-null   int64  
 5   lat                           5162 non-null   float64
 6   long                          5162 non-null   float64
 7   status                        5162 non-null   object 
 8   category                      5162 non-null   float64
 9   wind                          5162 non-null   int64  
 10  pressure                      5162 non-null   int64  
 11  tropicalstorm_force_diameter  5162 non-null   int64  
 12  hurricane_force_diameter      5162 non-null   int64  
 13  hurric